table/server/master/src/main/java/alluxio/master/table/Table.java

```diff
@@ -45,6 +49,15 @@
   private String mOwner;
   private List<ColumnStatisticsInfo> mStatistics;
   private Map<String, String> mParameters;
+  private Map<String, Partition> mSpecToPartitions;
+
+  private void computeSpecToPartitions(PartitionScheme partitionScheme) {
```

Comment:
I think this functionality should be owned by the `PartitionScheme`, so `PartitionScheme` can expose an api like `get(spec)`.

Reply:
added this interface API and abstracted out a `BasePartitionScheme` to provide default implementation